x-pack/plugins/enterprise_search/server/routes/app_search/engines.ts

```diff
@@ -44,6 +44,24 @@ export function registerEnginesRoutes({
     }
   );
 
+  router.post(
+    {
+      path: '/api/app_search/engines',
+      validate: {
+        body: schema.object({
+          name: schema.string(),
+          language: schema.maybe(schema.string()),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      return enterpriseSearchRequestHandler.createRequest({
+        path: `/as/engines`,
+        body: request.body,
+      })(context, request, response);
+    }
```

Comment:
No need to pass `body` to the `enterpriseSearchRequestHandler`, it already gets that from `request` and implicitly passes along all request bodies as-is. So we can simplify this to:

```suggestion
    enterpriseSearchRequestHandler.createRequest({
      path: '/as/engines',
    })
```